lib/drivers/medtronic600/NGPHistoryParser.js

```diff
@@ -1743,36 +1739,40 @@ class NGPHistoryParser {
         if (event.bolusPart === NGPUtil.NGPConstants.DUAL_BOLUS_PART.NORMAL_BOLUS) {
           const matchingBolus = this.findMatchingDualBolusEvent(event);
+          // Only upload an extended bolus once it is complete. We don't upload an
+          // in-progress bolus.
+          if (!_.isUndefined(matchingBolus)) {
+            const bolus = this.cfg.builder.makeDualBolus()
+              .with_normal(event.deliveredAmount)
+              .with_extended(matchingBolus.deliveredAmount)
+              .with_duration(matchingBolus.deliveredDuration * sundial.SEC_TO_MSEC);
+
+            if (event.normalProgrammedAmount !== event.deliveredAmount) {
+              bolus
+                .with_expectedNormal(event.normalProgrammedAmount);
+            }
 
-          const bolus = this.cfg.builder.makeDualBolus()
-            .with_normal(event.deliveredAmount)
-            .with_extended(matchingBolus.deliveredAmount)
-            .with_duration(matchingBolus.deliveredDuration * sundial.SEC_TO_MSEC);
-
-          if (event.normalProgrammedAmount !== event.deliveredAmount) {
-            bolus
-              .with_expectedNormal(event.normalProgrammedAmount);
-          }
-
-          if (matchingBolus.squareProgrammedAmount !== matchingBolus.deliveredAmount ||
-            matchingBolus.programmedDuration !== matchingBolus.deliveredDuration) {
-            bolus
-              .with_expectedExtended(matchingBolus.squareProgrammedAmount)
-              .with_expectedDuration(matchingBolus.programmedDuration * sundial.SEC_TO_MSEC);
-          }
+            if (matchingBolus.squareProgrammedAmount !== matchingBolus.deliveredAmount ||
+              matchingBolus.programmedDuration !== matchingBolus.deliveredDuration) {
+              bolus
+                .with_expectedExtended(matchingBolus.squareProgrammedAmount)
+                .with_expectedDuration(matchingBolus.programmedDuration * sundial.SEC_TO_MSEC);
+            }
 
-          this.addTimeFields(bolus, beginTimestamp);
-
-          if (event.isWizardBolus()) {
-            // Use parseFloat(num.toFixed(3)), because in JavaScript,
-            // 5.20 + 5.15 = 10.350000000000001
-            const wizardEvent = this.findWizardForBolus(
-              event,
-              parseFloat((event.normalProgrammedAmount + event.squareProgrammedAmount).toFixed(3)),
-            );
-            events.push(this.buildWizardRecord(wizardEvent, bolus.done()));
-          } else {
-            events.push(bolus.done());
+            this.addTimeFields(bolus, beginTimestamp);
+
+            if (event.isWizardBolus()) {
+              // Use parseFloat(num.toFixed(3)), because in JavaScript,
+              // 5.20 + 5.15 = 10.350000000000001
+              const wizardEvent = this.findWizardForBolus(
+                event,
+                parseFloat((event.normalProgrammedAmount + event.squareProgrammedAmount)
+                  .toFixed(3)),
```

Comment:
I usually use 5 significant digits, as I think that's what we use elsewhere as well. Or is there a specific reason for using three digits here?

Reply:
No particular reason. I didn't realise that 5 was used elsewhere.